qcodes/dataset/data_set_cache.py

```diff
@@ -49,13 +50,20 @@ def load_data_from_db(self) -> None:
         If the dataset is marked completed and data has already been loaded
         no load will be performed.
         """
+        if self.live:
+            raise RuntimeError("Cannot read data into this cache from the "
+                               "database as the dataset is build "
+                               "directly from measured data.")
```

Comment:
would this be a clearer message? or do we not want to mention "in memory" at all?
```suggestion
            raise RuntimeError("Cannot load data into this cache from the "
                               "database because this dataset is being built "
                               "in-memory.")
```